src/sentry/pipeline/__init__.py

```diff
@@ -146,6 +155,7 @@ def __init__(self, request, organization, provider_key, provider_model=None, con
         self.provider_model = provider_model
 
         self.config = config
+        self.provider.set_pipeline(self)
```

Comment:
this is definitely why the tests are failing. @MeredithAnya tomorrow maybe you can try removing this change and seeing what breaks for GHE? 

and/or @EvanPurkhiser do you by chance have any context into why this change would be necessary?

Reply:
This is for sure needed so that the provider is able to lookup the OAuth configuration from the parent pipleine (available on the executing pipeline). I'll take a look at tests.

Reply:
Should be fixed. The tests overrode the variable this assigned to.